.github/workflows/tagBasedImageBuild.yml

```diff
@@ -1,14 +1,24 @@
 name: Tag Based Image Build
 
 on:
-  create: # This listens to create events, which includes tag creations
+  create:
 
 jobs:
   buildImageForNewTag:
     if: startsWith(github.ref, 'refs/tags/') # Only run this job when a tag is created
     runs-on: ubuntu-latest
 
     steps:
+      - name: Extract tag name
+        id: get_tag_name
+        run: echo "TAG_NAME=${GITHUB_REF#refs/tags/}" >> $GITHUB_ENV
+
+      - name: Determine Branch Name
+        id: get_branch
+        run: |
+          branch=$(git branch -r --contains tags/${{ env.TAG_NAME }} | grep 'origin/' | sed 's|origin/||' | head -n 1)
+          echo "BRANCH_NAME=$branch" >> $GITHUB_ENV
+
```

Comment:
We need these steps here, as the `create` event, doesn't include the `tag_name` & `branch_name` which we need in the successive steps